sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/InstrumentBuilder.java

```diff
@@ -15,95 +16,72 @@
 import io.opentelemetry.sdk.metrics.internal.state.SdkObservableMeasurement;
 import io.opentelemetry.sdk.metrics.internal.state.WriteableMetricStorage;
 import java.util.Collections;
+import java.util.List;
 import java.util.function.BiFunction;
 import java.util.function.Consumer;
+import javax.annotation.Nullable;
 
 /** Helper to make implementing builders easier. */
-abstract class AbstractInstrumentBuilder<BuilderT extends AbstractInstrumentBuilder<?>> {
-
-  static final String DEFAULT_UNIT = "";
+final class InstrumentBuilder {
 
+  private final String name;
   private final MeterProviderSharedState meterProviderSharedState;
-  private final InstrumentType type;
-  private final InstrumentValueType valueType;
-  private String description;
-  private String unit;
-
-  protected final MeterSharedState meterSharedState;
-  protected final String instrumentName;
-  protected final Advice.AdviceBuilder adviceBuilder;
-
-  AbstractInstrumentBuilder(
-      MeterProviderSharedState meterProviderSharedState,
-      MeterSharedState meterSharedState,
-      InstrumentType type,
-      InstrumentValueType valueType,
+  private final MeterSharedState meterSharedState;
+  @Nullable private InstrumentType type = null;
+  @Nullable private InstrumentValueType valueType;
+  private Advice.AdviceBuilder adviceBuilder = Advice.builder();
+  private String description = "";
+  private String unit = "";
+
+  InstrumentBuilder(
       String name,
-      String description,
-      String unit) {
-    this(
-        meterProviderSharedState,
-        meterSharedState,
-        type,
-        valueType,
-        name,
-        description,
-        unit,
-        Advice.builder());
-  }
-
-  AbstractInstrumentBuilder(
       MeterProviderSharedState meterProviderSharedState,
-      MeterSharedState meterSharedState,
-      InstrumentType type,
-      InstrumentValueType valueType,
-      String name,
-      String description,
-      String unit,
-      Advice.AdviceBuilder adviceBuilder) {
-    this.type = type;
-    this.valueType = valueType;
-    this.instrumentName = name;
-    this.description = description;
-    this.unit = unit;
+      MeterSharedState meterSharedState) {
+    this.name = name;
     this.meterProviderSharedState = meterProviderSharedState;
     this.meterSharedState = meterSharedState;
-    this.adviceBuilder = adviceBuilder;
   }
 
-  protected abstract BuilderT getThis();
-
-  public BuilderT setUnit(String unit) {
+  public InstrumentBuilder setUnit(String unit) {
```

Comment:
#nit: I know this didn't change in this PR, but I believe all these setters can be package private.